pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -161,6 +161,16 @@ func createTargetConfigController_v311_00_to_latest(ctx context.Context, syncCtx
 		return true, err
 	}
 
+	if deprecatedPolicy {
+		upgradeableCondition := operatorv1.OperatorCondition{
+			Type:   operatorv1.OperatorStatusTypeUpgradeable,
+			Status: operatorv1.ConditionFalse,
```

Comment:
This should be similar to here: https://github.com/openshift/cluster-kube-apiserver-operator/blob/5435d65b520191ba74fb2cd8c84a4f5bdf604178/pkg/operator/certrotationtimeupgradeablecontroller/certrotationtime_upgradeable.go#L68-L71 so that we have clear information what caused it and how to prevent it. I'd imagine:
```golang
upgradeableCondition := operatorv1.OperatorCondition{
    Type: "PolicyUpgradeable",
    Status:  operatorv1.ConditionFalse,
	Reason:  "PolicyFieldSpecified",
    Message: fmt.Sprintf("scheduler.policy field is set to point to %q configmap which was deprecated and is to be removed in the next release", policy.Name),
}
```
or something like that :wink:

Reply:
Updated to your example